src/Microsoft.AspNetCore.Routing/Matching/RefEmitTrieJumpTable.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+#if NET462 || NETCOREAPP2_0
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+
+namespace Microsoft.AspNetCore.Routing.Matching
+{
+    internal class RefEmitTrieJumpTable : JumpTable
+    {
+        private readonly int _defaultDestination;
+        private readonly int _exitDestination;
+        private readonly (string text, int destination)[] _entries;
+
+        private readonly Func<string, int, int, int> _delegate;
+
+        public RefEmitTrieJumpTable(
+            int defaultDestination,
+            int exitDestination,
+            (string text, int destination)[] entries)
+        {
+            _defaultDestination = defaultDestination;
+            _exitDestination = exitDestination;
+            _entries = entries;
+
+            _delegate = GenerateDelegate();
+        }
+
+        public override int GetDestination(string path, PathSegment segment)
+        {
+            if (path.Length == 0)
+            {
+                return _exitDestination;
+            }
+
+            return _delegate(path, segment.Start, segment.Length);
+        }
+
+        private Func<string, int, int, int> GenerateDelegate()
+        {
+#if NET462
+            var assemblyName = new AssemblyName("DynamicRoutingJumpTable");
```

Comment:
Why a dynamic assembly over a dynamic method?

Reply:
So he can debug the IL by dumping it to a file 😄

Reply:
Yeah, I don't think we actually want to cross-target for net46. I'll likely put this code in a separate tool that can dump the jump tables if you need to run peverify or something.